app/controllers/admin/elections_controller.rb

```diff
@@ -1,72 +1,75 @@
   end
-
   def create
-    if @election.save
-      redirect_to [:admin, @election], notice: 'Valet skapades'
-    else
-      render action: :new
+    @election = Election.new(election_params)
+    respond_to do |format|
+      if @election.save
+        format.html { redirect_to [:admin,@election], notice: 'Valet skapades!' }
+      else
+        format.html { render action: "new" }
+      end
     end
   end
-
   def update
-    if @election.update(election_params)
-      redirect_to [:admin, @election], notice: 'Valet uppdaterades, gött'
-    else
-      render action: :edit
+    respond_to do |format|
+      if @election.update(election_params)
+        format.html { redirect_to admin_election_path(@election), notice: 'Valet uppdaterades, gött' }
+        format.json { head :no_content }
+      else
+        format.html { render action: 'edit' }
+        format.json { render json: @election.errors, status: :unprocessable_entity }
+      end
     end
   end
 
   def destroy
     @election.destroy
-    redirect_to admin_elections_path, notice: 'Valet raderades, hoppas att det var meningen!.'
+    respond_to do |format|
+      format.html { redirect_to elections_path,notice: 'Valet raderades, hoppas att det var meningen!.' }
+      format.json { head :no_content }
+    end
   end
 
   def nominations
     @nominations_grid = initialize_grid(@election.nominations,
-                                        name: 'g2',
-                                        enable_export_to_csv: true,
-                                        csv_file_name: 'nomineringar')
+                                        :name => 'g2',
+                                        :enable_export_to_csv => true,
+                                        :csv_file_name => 'nomineringar')
   end
-
   def candidates
-    @candidates_grid = initialize_grid(@election.candidates,
-                                       name: 'candidates',
-                                       enable_export_to_csv: true,
-                                       csv_file_name: 'candidates')
+    @candidates_grid = initialize_grid(@election.candidates,name: "candidates",enable_export_to_csv: true, csv_file_name: 'candidates')
     export_grid_if_requested('g2' => 'nominations_grid')
   end
-
   private
-
   def authorize
-    authorize! :manage, Election
+    authorize! :manage, @election
   end
 
   def election_params
-    params.fetch(:election).permit(:title, :description, :start, :end, :url,
-                                   :visible, :mail_link, :mail_styrelse_link, :text_before,
-                                   :text_during, :text_after, :nominate_mail, :candidate_mail,
-                                   :extra_text, :candidate_mail_star, post_ids: [])
+    params.fetch(:election).permit(:title,:description,:start,:end,:url,:visible,:mail_link,:mail_styrelse_link,:text_before,:text_during,:text_after,:nominate_mail,:candidate_mail,:extra_text,:candidate_mail_star,:post_ids => [])
```

Comment:
Line is too long. [230/100]<br>Use the new Ruby 1.9 hash syntax.<br>Space missing after comma.